### PR TITLE
kaiax/gov: Serialize/deserialize validator votes

### PR DESCRIPTION
**Pull request by ian0371**
## Proposed changes

- Facilitates `governance_vote` to support validator votes (i.e., `governance.addvalidator`/`governance.removevalidator`).
  - Adds address-list encoding in `ToVoteBytes`.
  - Adds address-list canonicalizer `validatorAddressListCanonicalizer`.
- Adds `ValidatorParams` for checking validator-related votes.
- Adds testcases from real blocks: `TestVoteSerialization`.

## Types of changes

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist
- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/kaiachain/kaia/blob/main/CONTRIBUTING.md) doc
- [x] I have read the [CLA](https://gist.github.com/kaiachain-dev/bbf65cc330275c057463c4c94ce787a6) and signed by comment `I have read the CLA Document and I hereby sign the CLA` in first time contribute
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

This PR fixes encoding of multiple-address votes, because the previous scheme involves decoding RLP twice.
This data does not exist in any on-chain blocks, so there's no need to support backward compatibility.

```
// before this PR
rlp decode 0xf85b94f39fd6e51aad88f6f4ce6ab8827279cfffb922669a676f7665726e616e63652e72656d6f766576616c696461746f72ea943c44cdddb6a900fa2b585dd299e03d12fa4293bc9490f79bf6eb2c4f870365e785982e1f101e93b906
["f39fd6e51aad88f6f4ce6ab8827279cfffb92266","676f7665726e616e63652e72656d6f766576616c696461746f72",["3c44cdddb6a900fa2b585dd299e03d12fa4293bc","90f79bf6eb2c4f870365e785982e1f101e93b906"]]

// after this PR
rlp decode 0xf85994f39fd6e51aad88f6f4ce6ab8827279cfffb922669a676f7665726e616e63652e72656d6f766576616c696461746f72a83c44cdddb6a900fa2b585dd299e03d12fa4293bc90f79bf6eb2c4f870365e785982e1f101e93b906
["f39fd6e51aad88f6f4ce6ab8827279cfffb92266","676f7665726e616e63652e72656d6f766576616c696461746f72","3c44cdddb6a900fa2b585dd299e03d12fa4293bc90f79bf6eb2c4f870365e785982e1f101e93b906"]
```

### CODE REVIEW

File: kaiax/gov/param.go [hunk 1]
```diff
@@ -51,11 +53,45 @@ var (
...
 
 		switch v := v.(type) {
-		case []byte:
-			return stringToAddressList(string(v))
-		case string:
+		case []byte: // input from header.Vote
+			// There are three types of header.Vote encoding for validator address(es).
+			// Type1. Single address, [20]byte. See Mainnet block 5505383.
+			// Type2. Multiple addresses, [20*n]byte.
+			// Type3. Single address, [42]byte (hex-encoded bytes). See Mainnet block 90915008.
```
Comment (blukat29) [inline kaiax/gov/param.go:60]: Are you going to make it valid? It could have been a bug in governance_vote

Comment (ian0371) [reply kaiax/gov/param.go:60]: I think it was a bug. Where do you think is appropriate if not here?

File: kaiax/gov/headergov/impl/init.go [hunk 2]
```diff
@@ -154,8 +155,12 @@ func (h *headerGovModule) scanAllVotesInEpoch(epochIdx uint64) map[uint64]header
 			logger.Error("Failed to parse vote", "num", blockNum, "err", err)
 			continue
 		}
-		// TODO-kaiax: consider writing addval/removeval votes to validator DB.
-		if vote != nil && vote.Name() != "governance.addvalidator" && vote.Name() != "governance.removevalidator" {
+
+		if vote == nil {
```
Comment (blukat29) [inline kaiax/gov/headergov/impl/init.go:159]: Will addval/removeval vote blocks be stored in VoteDataBlockNum?

File: kaiax/gov/param.go (hunk 1 above)
Comment (blukat29) [reply kaiax/gov/param.go:60]: We could ignore such vote I think? currently you are parsing the data without problem.

Comment (ian0371) [reply kaiax/gov/param.go:60]: Without type3 parsing logic here, `VerifyHeader()` will fail during sync test.
With this PR, `NewVoteData()` should return a canonicalized vote data on validator votes. (Previously, it returned `voteData` without value)

File: kaiax/gov/headergov/impl/init.go (hunk 2 above)
Comment (ian0371) [reply kaiax/gov/headergov/impl/init.go:159]: No. This check is moved to below lines.

File: kaiax/gov/param.go (hunk 1 above)
Comment (blukat29) [reply kaiax/gov/param.go:60]: I see. Then it accepts hex-encoded address as valid vote, but will not produce anymore via governance_vote.

File: kaiax/gov/headergov/impl/init.go (hunk 2 above)
Comment (ian0371) [reply kaiax/gov/headergov/impl/init.go:159]: Done 613ad27b

**Review by hyunsooda — APPROVED** [review]

**Review by blukat29 — APPROVED** [review]

**Pull request merged**